src/gateway/kong-production/monitoring/statsd.md

```diff
@@ -1,6 +1,108 @@
 ---
-title: Monitoring with Statsd
-
+title: Monitoring with StatsD
+content_type: how-to
 ---
 
-## PLACEHOLDER How to Monitor with Statsd
\ No newline at end of file
+[StatsD](https://github.com/statsd/statsd) is a network daemon that collects
+and aggreates performance metrics by listening on the network for simple 
+text based statistics data published by applications. 
+
+This guide will help you setup a test {{site.base_gateway}} and
+StatsD service. Then you will generate sample requests to {{site.base_gateway}} and
+observe the collected monitoring data.
+
+### Prerequisites
+This guide assumes each of the following tools are installed locally. 
+* [Docker](https://docs.docker.com/get-docker/) is used to run Kong, the supporting database, and StatsD locally. 
+* [curl](https://curl.se/) is used to send requests to the gateway. Most systems come with `curl` pre-installed.
+* [Netcat](http://netcat.sourceforge.net/) installed as `nc` on the `PATH`. `nc` is used to send requests 
+  to the StatsD management interface. Many systems come with `nc` pre-installed.
+
+### Guide
+
+1. Run the following command to start {{site.base_gateway}} using Docker:
```

Comment:
@rspurgeon What happens if you already have Gateway installed and you try to run this script? I was kind of thinking that the audience of this doc might already have Gateway installed. If so, would it be better to tell them which routes/services/etc. to configure so they can use this guide? Or we could provide the option like:

1. Do one of the following:
    - If you don't have Kong installed, run this quickstart script:
    - If you've already installed Kong, make sure you add these services/routes that we'll be using for this guide:

Reply:
The quickstart script is designed to rebuild the test gateway if it already exists.

IMO we should just give the user the necessary instructions inline of page.  It's a single line of code to get the test gateway, so i think it's a distraction to redirect them to another page.  The script also sets up the mock service which aids in the later instructions, otherwise we have to write confusing "if you have this do this if you don't have this do that" kind of copy.

Reply:
I think Diana's idea is a good one. There's nothing wrong with having the script as a prerequisite, or providing the alternative to add a list of routes and services that they will be using in this guide. 

While the quickstart script and the entities it creates are great resources, sticking this in as an absolute requirement gates the guide for anyone with an existing installation. 

> otherwise we have to write confusing "if you have this do this if you don't have this do that" kind of copy.

We can write this once, in the prereq, and then the rest of the guide can just assume they have the routes/services.

> The quickstart script is designed to rebuild the test gateway if it already exists.

That seems like a bad idea, if the person isn't expecting it. If we don't give them the option to add this to their existing installation, then we're losing potential users.

As much as I like the idea of "every page is page one", starting with an installation of kong gateway for a guide like this is going too far. We have to remember that our audience is varied, and users of all stages in their Kong journey could be wanting to do something like set up statsd - by adding a step to install gateway anew, we're essentially saying "you can't actually do this in your installation, you have to start over".

Reply:
In my experience this kind of soup-to-nuts quickstart documentation is best for getting up and running with concepts, otherwise the documentation becomes littered with conditionals for the reader (click this link if you have this or do this if you don't).  Documentation pages that are simple, complete, and to the point is the best in my personal opinion.

The task at hand is to learn about the StatsD plugin and the concepts around it.  As a potential reader of this documentation, I would assume I could apply the techniques to my own installation, or else why would it exist at all? 

This is why people use Docker for development, ephemeral things to test that are easily disposed and recreated. The script gives them a quick test environment to work with, named in a particular way that can be disposed of with no cost.  They should have no expectation about it that would surprise them.  

When testing new concepts, it's less common to want to add to an existing service, one that is likely already serving another purpose. You spin up something new to learn the objective, take the education and then apply it. I wouldn't expect lose any potential users over this workflow.  

All that said, I think some dialog at the beginning of the document that says, "If you don't have a Kong Gateway, run this script" and keep the script instruction _in line of the page_ would be fine.

Then we also need to say something like... "If you're using an existing gateway, modify the connection information in the following command."  Otherwise we need to use a variable or something which is currently represented in the doc as `localhost:8000-1`.   As Diana also mentioned, we'd also need to add something to inform the user to add a new route and service to match the document, or tell them to modify the commands as needed.

In my personal opinion, it adds a lot of extra cognitive overhead for the reader who is very likely learning this on a test gateway they will create. Regardless, please feel free to make the desired changes to the document as the team desires to get it complete for the release.